src/Modelling/ActivityDiagram/SelectAS.hs

```diff
@@ -241,9 +242,10 @@ generateCorrectSequenceWithDuplication baseSequence ad = do
         ]
 
       -- Strategy 2: Duplicate subsequences of the base sequence
+      -- Try all subsequence lengths from 2 up to the full sequence length to handle cycles of any size
       subsequenceDuplicates =
         [ baseSequence ++ subsequence
-        | sequenceLength <- [2, 3]  -- Try subsequences of length 2 and 3
+        | sequenceLength <- [2..maxSubsequenceLength]  -- Try all subsequence lengths
         , sequenceLength <= length baseSequence
         , startIndex <- [0..length baseSequence - sequenceLength]
```

Comment:
Since `maxSubsequenceLength = length baseSequence`, the preceding two lines here should be simplified.